bolt-sidecar/src/api/commitments/handlers.rs

```diff
@@ -26,6 +28,34 @@ use super::{
     },
 };
 
+/// Response structure for the metadata endpoint that combines
+/// limits configuration with version information in a flat structure
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MetadataResponse {
+    /// Max number of commitments to accept per slot
+    pub max_commitments_per_slot: NonZero<usize>,
+    /// Max committed gas per slot
+    pub max_committed_gas_per_slot: NonZero<u64>,
+    /// Min priority fee to accept for a commitment
+    pub min_priority_fee: u128,
+    /// The maximum size in MiB of the account states
+    pub max_account_states_size: NonZero<usize>,
+    /// The version of the Bolt sidecar
+    pub version: String,
+}
```

Comment:
Tried to use flatten instead, but had problems deserializing the response in the test because `serde_json` doesn't support `u128`

```rust
pub struct MetadataResponse<T> {
    /// The operational limits of the sidecar
    #[serde(flatten)]
    pub limits: LimitOp,
    /// The version of the Bolt sidecar
    pub version: String,
}
```

Reply:
I don't remember, is it necessary to use `u128` tho? If in wei, a `u64` can express a `min_priority_fee` of `(2**64 - 1) / 10**18 == 18 ETH`, should be enough :D

Reply:
oof, I don't love that tbh! Why put this limit in the code? let's either find a type that works like alloy::primitives::U128 or what @estensen did